src/main/java/microsoft/exchange/webservices/data/ServiceRequestBase.java

```diff
@@ -621,13 +621,24 @@ protected HttpWebRequest validateAndEmitRequest() throws ServiceLocalException,
     this.validate();
 
     HttpWebRequest request = this.buildEwsHttpWebRequest();
+
     try {
-      return this.getEwsHttpWebResponse(request);
-    } catch (HttpErrorException e) {
-      processWebException(e, request);
+      try {
+        return this.getEwsHttpWebResponse(request);
+      } catch (HttpErrorException e) {
+        processWebException(e, request);
 
-      // Wrap exception if the above code block didn't throw
-      throw new ServiceRequestException(String.format(Strings.ServiceRequestFailed, e.getMessage()), e);
+        // Wrap exception if the above code block didn't throw
+        throw new ServiceRequestException(String.format(Strings.ServiceRequestFailed, e.getMessage()), e);
+      }
+    } catch (Exception e) {
```

Comment:
Because the request should be closed on every Exception (because the request is not returned in that case), including the ones raised in `processWebException()`. The exception handling in this libary is very messy, unfortunately.